uk.org.greenend.chiark.sgtatham.puzzles.yaml

```diff
@@ -0,0 +1,48 @@
+app-id: uk.org.greenend.chiark.sgtatham.puzzles 
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: mosaic
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  # Needs to save files locally - not supporting portals yet
+  - --filesystem=xdg-documents
+modules:
+# from https://github.com/scx/xaralx-flatpak/blob/master/imagemagick.yaml
+  - name: imagemagick
+    config-opts:
+      - --enable-shared
+      - --disable-static
+      - --disable-docs
+      - --with-modules
+      - --with-x
+      - --with-threads
+      - --with-magick_plus_plus
+      - --with-gslib
+      - --with-openexr
+      - --with-rsvg
+      - --with-xml
+      - --with-jbig
+      - --with-openjp2
+    sources:
+      - type: archive
+        url: https://github.com/ImageMagick/ImageMagick6/archive/6.9.10-23.tar.gz
+        sha256: 2a0b11952c994b06562dace16a53c01bdd0f715de59483145eb3053902a42361
+    # We need it only for the build stage
+    cleanup:
+      - '*'
+  - name: puzzles
+    buildsystem: cmake-ninja
+    post-install:
+      - install -Dm644 uk.org.greenend.chiark.sgtatham.puzzles.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - install -Dm644 icons/mosaic-128d24.png ${FLATPAK_DEST}/share/app-info/icons/flatpak/128x128/uk.org.greenend.chiark.sgtatham.puzzles.png
```

Comment:
```suggestion
      - install -Dm644 icons/mosaic-128d24.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/uk.org.greenend.chiark.sgtatham.puzzles.png
```

Reply:
It failed because of this change, it seems: `test -f builddir/*/share/app-info/icons/flatpak/128x128/uk.org.greenend.chiark.sgtatham.puzzles.png'`

Reply:
no its not. It's missing the desktop file.

Reply:
It needs a desktop file named `uk.org.greenend.chiark.sgtatham.puzzles.desktop`.

Reply:
This install path you use is **WRONG**.